web/api/v1/api.go

```diff
@@ -250,7 +267,11 @@ func (api *API) query(r *http.Request) (interface{}, *apiError, func()) {
 		defer cancel()
 	}
 
-	qry, err := api.QueryEngine.NewInstantQuery(api.Queryable, r.FormValue("query"), ts)
+	newQueryable := api.Queryable
+	if fanout, ok := newQueryable.(storage.FanoutStorage); ok {
```

Comment:
Using a factory function for api.Queryable would seem cleaner than checking the type here

Reply:
I'm not quite following, do you want to add a function to the Queryable interface that would do this?